File: java/Program_3.java
import java.util.Scanner;

//Program-3 :-  Write a program to find the i-th bit.
public class Program_3 {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter a number");
        int n=sc.nextInt();
        System.out.println("Enter bit number");
        int i=sc.nextInt();
        
        if((n&(1<<i))==0){
            System.out.println("The "+i+"th bit is 0");
        }else{
            System.out.println("The "+i+"th bit is 1");
        }

    }
}

File: c++/Program-3.cpp
//Program-3     :- Write a program to find the i th bit.

#include<iostream>
using namespace std;

int main(){
    int num,pos;
    cin>>num>>pos;
    if((num&(1<<pos))==0){
        cout<<pos<<" bit is 0 \n";
    }else{
        cout<<pos<<" bit is 1 \n";
    }
    return 0;
}